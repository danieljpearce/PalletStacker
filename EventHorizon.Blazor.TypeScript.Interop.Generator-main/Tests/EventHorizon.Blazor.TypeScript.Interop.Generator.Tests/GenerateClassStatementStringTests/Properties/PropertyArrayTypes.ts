declare module Static.Primitive {
    export class ExampleClass {
        arrayExample: Array<string>;
        arrayLikeExample: ArrayLike<string>;
        floatArrayExample: FloatArray;
        float32ArrayExample: Float32Array;
        indicesArrayExample: IndicesArray;
        uint8ArrayExample: Uint8Array;
    }
}
