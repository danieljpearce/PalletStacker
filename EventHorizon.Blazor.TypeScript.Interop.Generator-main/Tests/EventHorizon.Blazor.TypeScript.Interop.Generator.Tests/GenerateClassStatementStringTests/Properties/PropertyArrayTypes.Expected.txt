/// Generated - Do Not Edit
namespace Static.Primitive
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties
        
        public string[] arrayExample
        {
            get
            {
            return EventHorizonBlazorInterop.GetArray<string>(
                    this.___guid,
                    "arrayExample"
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "arrayExample",
                    value
                );
            }
        }

        
        public string[] arrayLikeExample
        {
            get
            {
            return EventHorizonBlazorInterop.GetArray<string>(
                    this.___guid,
                    "arrayLikeExample"
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "arrayLikeExample",
                    value
                );
            }
        }

        
        public decimal[] floatArrayExample
        {
            get
            {
            return EventHorizonBlazorInterop.GetArray<decimal>(
                    this.___guid,
                    "floatArrayExample"
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "floatArrayExample",
                    value
                );
            }
        }

        
        public decimal[] float32ArrayExample
        {
            get
            {
            return EventHorizonBlazorInterop.GetArray<decimal>(
                    this.___guid,
                    "float32ArrayExample"
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "float32ArrayExample",
                    value
                );
            }
        }

        
        public decimal[] indicesArrayExample
        {
            get
            {
            return EventHorizonBlazorInterop.GetArray<decimal>(
                    this.___guid,
                    "indicesArrayExample"
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "indicesArrayExample",
                    value
                );
            }
        }

        
        public decimal[] uint8ArrayExample
        {
            get
            {
            return EventHorizonBlazorInterop.GetArray<decimal>(
                    this.___guid,
                    "uint8ArrayExample"
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "uint8ArrayExample",
                    value
                );
            }
        }
        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}